### PR TITLE
CLI: test validation; local repo settings

### LINKED ISSUE #17927: Quarkus CLI create app to move to created directory
## Description

Currently, when you create an application in either batch mode or normal mode the current directory remains the same. From a user perspective, it can be assumed that after you have created your app you want to do additional quarkus CLI tasks like add an extension or build the project so it would feel more natural if the Quarkus CLI in interactive mode changed the current directory to the newly created application.

## Implementation ideas

I think it makes sense to move to the directory of the newly created application in interactive mode, while it remains in the current directory in batch mode.

### LINKED ISSUE #18285: quarkus build -Dkey=value pairs should be passed through blindly
## Describe the bug
at the moment we have special "propagatePropertyIfset" for quarkus build specific properties:

maven.repo.local, io.quarkus.maven.secondary-local-repo  and CONFIG_FILE_PATH_PROPERTY

this should not be in the cli as will have weird sidefects onusers.

### Expected behavior 
just pass any -Dkey=value through. 
I can;t think of a case where the cli treats key=value as somethng special that the build system shouldn't know about.

### PR DESCRIPTION
**Pull request by ebullient**
Resolves #18285 
Resolves #17927

### CODE REVIEW

**Comment by gsmet** [comment]
Is it something we should backport to 2.0?

File: devtools/cli/src/main/java/io/quarkus/cli/Completion.java [hunk 1]
```diff
@@ -4,12 +4,7 @@
 import picocli.CommandLine;
 
 @CommandLine.Command(name = "completion", version = "generate-completion "
-        + CommandLine.VERSION, mixinStandardHelpOptions = true, description = {
-                "bash/zsh completion:  source <(${PARENT-COMMAND-FULL-NAME:-$PARENTCOMMAND} ${COMMAND-NAME})",
-                "Run the following command to give `${ROOT-COMMAND-NAME:-$PARENTCOMMAND}` TAB completion in the current shell:",
-                "",
-                "  source <(${PARENT-COMMAND-FULL-NAME:-$PARENTCOMMAND} ${COMMAND-NAME})",
-                "" }, optionListHeading = "Options:%n", helpCommand = true)
+        + CommandLine.VERSION, mixinStandardHelpOptions = true, header = "bash/zsh completion:  source <(${PARENT-COMMAND-FULL-NAME:-$PARENTCOMMAND} ${COMMAND-NAME})", helpCommand = true, headerHeading = "%n", commandListHeading = "%nCommands:%n", synopsisHeading = "%nUsage: ", optionListHeading = "Options:%n")
```
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/Completion.java:7]: are this introduction of *Heading attributes just to create bit more "air" with extra newlines or ?

File: devtools/cli/src/main/java/io/quarkus/cli/ProjectExtensionsList.java [hunk 2]
```diff
@@ -18,35 +18,35 @@
...
+    @CommandLine.ArgGroup(order = 2, heading = "%nQuarkus version (absolute):%n")
     TargetQuarkusVersionGroup targetQuarkusVersion = new TargetQuarkusVersionGroup();
 
     @CommandLine.Option(names = { "-i",
             "--installable" }, defaultValue = "false", order = 2, description = "List extensions that can be installed (relative)")
     boolean installable = false;
 
     @CommandLine.Option(names = { "-s",
-            "--search" }, defaultValue = "*", paramLabel = "PATTERN", order = 3, description = "Search filter on extension list (Java Pattern syntax).")
+            "--search" }, defaultValue = "*", paramLabel = "PATTERN", order = 3, description = "Search filter on extension list (java.util.regex.Pattern).")
```
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/ProjectExtensionsList.java:42]: if it is java.util.regex.Pattern should the default not be .* then ? * alone is not a valid java util regex pattern.

File: devtools/cli/src/main/java/io/quarkus/cli/Completion.java (hunk 1 above)
Comment (ebullient) [reply devtools/cli/src/main/java/io/quarkus/cli/Completion.java:7]: Yes.

**Comment by ebullient** [comment]
ok. Done fussing. @maxandersen @mkouba  .. if you want to have a look

File: devtools/cli/src/main/java/io/quarkus/cli/CreateApp.java [hunk 3]
```diff
@@ -70,12 +73,18 @@ public Integer call() throws Exception {
...
+            } else if (buildTool == BuildTool.JBANG) {
                 success = new CreateJBangProjectCommandHandler().execute(invocation).isSuccess();
             } else { // maven or gradle
                 success = new CreateProjectCommandHandler().execute(invocation).isSuccess();
             }
-            return success ? CommandLine.ExitCode.OK : CommandLine.ExitCode.SOFTWARE;
+
+            if (success) {
+                output.info(
+                        "Navigate into this directory and get started: " + spec.root().qualifiedName() + " dev");
```
Comment (gsmet) [inline devtools/cli/src/main/java/io/quarkus/cli/CreateApp.java:84]: Why `dev` at the end?

File: devtools/cli/src/main/java/io/quarkus/cli/ProjectExtensionsList.java [hunk 4]
```diff
@@ -18,35 +18,35 @@
...
+    @CommandLine.ArgGroup(order = 2, heading = "%nQuarkus version (absolute):%n")
     TargetQuarkusVersionGroup targetQuarkusVersion = new TargetQuarkusVersionGroup();
 
     @CommandLine.Option(names = { "-i",
             "--installable" }, defaultValue = "false", order = 2, description = "List extensions that can be installed (relative)")
     boolean installable = false;
 
     @CommandLine.Option(names = { "-s",
-            "--search" }, defaultValue = "*", paramLabel = "PATTERN", order = 3, description = "Search filter on extension list (Java Pattern syntax).")
+            "--search" }, defaultValue = "*", paramLabel = "PATTERN", order = 3, description = "Search for matching extensions (simple glob using '*' and '?').")
```
Comment (gsmet) [inline devtools/cli/src/main/java/io/quarkus/cli/ProjectExtensionsList.java:42]: Ah cool, I wanted to suggest the use of a simple glob.

File: devtools/cli/src/main/java/io/quarkus/cli/CreateApp.java (hunk 3 above)
Comment (ebullient) [reply devtools/cli/src/main/java/io/quarkus/cli/CreateApp.java:84]: shows the dev command: `quarkus dev`

Comment (gsmet) [reply devtools/cli/src/main/java/io/quarkus/cli/CreateApp.java:84]: Ah OK, makes sense.

**Review by gsmet — APPROVED** [review]

**Pull request merged**